### PR TITLE
[Integration][Gitlab] | Fix unintended deletion of file kinds

### PR DESCRIPTION
**Pull request by mk-armah**
# Description

What - Addresses unintended deletion of file-kind entities by enforcing the use of advanced search functionality for project file lookups.

Why - Entities created with a file-kind type get unexpectedly deleted due to inconsistent search results.

How - This fix introduces an explicit search_type parameter set to advanced within project.search, ensuring that all relevant files are accurately located during searches, thereby preventing inadvertent deletions during resource updates.

## Type of change

Please leave one option from the following and delete the rest:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [x] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [x] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [x] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/gitlab/pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "gitlab"
-version = "0.1.134"
+version = "0.1.135-rc2"
```
Comment (Tankilevitch) [inline integrations/gitlab/pyproject.toml:3]:
```suggestion
version = "0.1.135-rc3"
```

File: integrations/gitlab/gitlab_integration/ocean.py [hunk 2]
```diff
@@ -185,36 +185,57 @@ async def resync_files(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
...
-
-                if tasks:
-                    logger.info(f"Found {len(tasks)} relevant projects in batch")
-                    async for batch in stream_async_iterators_tasks(*tasks):
-                        yield batch
-                else:
-                    logger.info("No relevant projects were found in batch, skipping it")
+        with logger.contextualize(
+            files_path=selector.files.path, files_repos=selector.files.repos
+        ):
```
Comment (Tankilevitch) [inline integrations/gitlab/gitlab_integration/ocean.py:198]: reminder to remove

Comment (mk-armah) [reply integrations/gitlab/gitlab_integration/ocean.py:198]: acknowledged

**Review by Tankilevitch — APPROVED** [review]
LGTM

File: integrations/gitlab/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 <!-- towncrier release notes start -->
 
+0.1.135 (2024-10-31)
+====================
+
+### Improvements
+
+- Resolved unexpected deletion of file kind entities by explicitly declaring the file search in projects to use the advanced search type.
```
Comment (Tankilevitch) [inline integrations/gitlab/CHANGELOG.md:15]:
```suggestion
-  Explicitly declaring the file search in projects to use the advanced search type, in cases where the default search in gitlab changes.
- Enhanced more verbosity on file kind
```

**Pull request merged**